catroid/src/main/java/org/catrobat/catroid/transfers/project/ProjectDownloadService.kt

```diff
@@ -0,0 +1,175 @@
+    }
+
+    override fun onHandleIntent(intent: Intent?) {
+        val downloadIntent = intent
+            ?: return logWarning("Called ProjectDownloadService with null intent - aborting")
+        val projectName = downloadIntent.getStringExtra(EXTRA_DOWNLOAD_NAME)
+            ?: return logWarning("Called ProjectDownloadService with null projectName -  aborting")
+        val url = downloadIntent.getStringExtra(EXTRA_URL)
+            ?: return logWarning("Called ProjectDownloadService without url - aborting")
+        val resultReceiver = downloadIntent.getParcelableExtra<ResultReceiver>(EXTRA_RESULT_RECEIVER)
+                ?: return logWarning("Called ProjectDownloadService without url - aborting")
+
+        val zipFileString = File(File(CACHE_DIR, TMP_DIR_NAME), DOWNLOAD_FILE_NAME).absolutePath
+        val destinationFile = File(zipFileString)
+
+        if ((destinationFile.parentFile.isDirectory or destinationFile.parentFile.mkdirs()).not()) {
+            ToastUtil.showError(this, R.string.error_project_download)
+            return
+        }
+
+        val id = downloadIntent.getIntExtra(EXTRA_NOTIFICATION_ID, 0)
+        val statusBarNotificationManager = StatusBarNotificationManager.getInstance()
+        val notification = statusBarNotificationManager.getNotification(this, id)
+
+        startForeground(id, notification)
+
+        CatrobatServerCalls(CatrobatWebClient.client)
+            .downloadProject(
+                url,
+                destinationFile,
+                object : DownloadSuccessCallback {
+                    override fun onSuccess() {
+                        downloadSuccessCallback(
+                            this@ProjectDownloadService,
+                            projectName,
+                            destinationFile,
+                            resultReceiver
+                        )
+                    }
+                },
+                object : DownloadErrorCallback {
+                    override fun onError(code: Int, message: String) {
+                        downloadErrorCallback(this@ProjectDownloadService, resultReceiver, projectName)
+                    }
+                },
+                object : DownloadProgressCallback {
+                    override fun onProgress(progress: Long) {
+                        downloadProgressCallback(this@ProjectDownloadService, resultReceiver, id, progress)
+                    }
+                }
+            )
+
+        stopForeground(true)
+    }
+
+    private fun downloadSuccessCallback(
+        context: Context,
+        projectName: String,
+        destinationFile: File,
+        resultReceiver: ResultReceiver
+    ) {
+        val statusBarNotificationManager = StatusBarNotificationManager.getInstance()
+        val notificationId = statusBarNotificationManager.createProjectDownloadNotification(this,
+                                                                                            projectName)
```

Comment:
this linebreak seems kind of funny :)

Reply:
Ups, must have been the auto-correct of android studio ^^